services/rclone-http/default.nix

```diff
@@ -0,0 +1,75 @@
+# SPDX-FileCopyrightText: 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  pkgs,
+  lib,
+  config,
+  ...
+}:
+with lib; let
+  cfg = config.services.rclone-http;
+
+  addr =
+    if cfg.withCaddy
+    then "unix://%t/rclone-http/socket"
+    else "localhost:8080";
+in {
+  options.services.rclone-http = {
+    enable = mkEnableOption "rclone-http service";
+
+    withCaddy = mkOption {
+      type = types.bool;
+      default = false;
+    };
+
+    readOnly = mkOption {
```

Comment:
add a description. Question, didn't check: we still need webdav and http if we also have this?